### PR TITLE
Adding projects for RecoveryServices.SiteRecovery service.

### PR DESCRIPTION
**Pull request by avneeshrai**
It's corresponding to new swagger spec which is under PR https://github.com/Azure/azure-rest-api-specs/pull/1209
Intent is to get early feedback here.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by avneeshrai** [comment]
Can someone review this PR please. Corresponding swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/1209 is merged.

I have removed do not merge tag.

File: src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs]: @avneeshrai missing license header.
Chances are your swagger does not have the right License Header Constant, please double check, I believe the constant that you need to use in your swagger is "MIT-no-version"

File: src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
```
Comment (shahabhijeet) [inline src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs]: @avneeshrai please use latest AutoRest (at least 1.0)

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs (hunk 1 above)
Comment (avneeshrai) [inline src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs]: @shahabhijeet can you please point to some example spec on what exacty needs to be done here
I tried searching MIT-no-version in azure-rest-api-spec repository but couldnt find any instance of it

File: src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs (hunk 2 above)
Comment (avneeshrai) [inline src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs]: even after doing autorest --latest or autorest --reset 
the version that i get is 0.16.0.0.
How to i get 1.0?

File: src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs (hunk 1 above)
Comment (avneeshrai) [inline src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs]: Fixed in the latest commit

**Comment by avneeshrai** [comment]
@shahabhijeet Please review and merge the change in swagger spec as you had suggested in the PR: https://github.com/Azure/azure-rest-api-specs/pull/1227

**Review by shahabhijeet — APPROVED** [review]

File: src/SDKs/RecoveryServices.SiteRecovery/RecoveryServices.SiteRecovery.Tests/RecoveryServices.SiteRecovery.Tests.csproj [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (avneeshrai) [inline src/SDKs/RecoveryServices.SiteRecovery/RecoveryServices.SiteRecovery.Tests/RecoveryServices.SiteRecovery.Tests.csproj]: @shahabhijeet as discussed offline...we dont have dependency on Resource manager in our projects. Have removed all the stale references that we had.

**Pull request merged**